server/http/src/service.rs

```diff
@@ -94,24 +122,41 @@ impl Service for GraphQLService {
     type Future = GraphQLServiceResponse;
 
     fn call(&mut self, req: Request<Self::ReqBody>) -> Self::Future {
-        match (req.method(), req.uri().path()) {
-            // GraphiQL
-            (&Method::GET, "/") => self.serve_file(include_str!("../assets/index.html")),
-            (&Method::GET, "/graphiql.css") => {
+        let method = req.method().clone();
+        let uri = req.uri().clone();
+        match (method, uri.path()) {
+            (Method::GET, "/") => self.index(),
+
+            (Method::GET, "/graphiql.css") => {
                 self.serve_file(include_str!("../assets/graphiql.css"))
             }
-            (&Method::GET, "/graphiql.min.js") => {
+            (Method::GET, "/graphiql.min.js") => {
                 self.serve_file(include_str!("../assets/graphiql.min.js"))
             }
 
-            // POST / receives GraphQL queries
-            (&Method::POST, "/graphql") => self.handle_graphql_query(req),
+            // Request is relative to a subgraph.
+            (method, path) => {
+                let mut path = path.split('/');
+                let _empty = path.next();
+                let name = path.next();
```

Comment:
As a consequence of handling both, names and IDs here, perhaps `name` here should be `subgraph` or `name_or_id`. But I think `subgraph` is ok.